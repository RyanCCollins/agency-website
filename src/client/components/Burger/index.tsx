import * as React from 'react';
import { SvgIcon } from 'ui';
import Button from './button';
import Span from './span';

export interface Props {
  onClick: React.EventHandler<React.SyntheticEvent<{}>>;
}
export default function Burger(props: Props): JSX.Element {
  return (
    <Button {...props}>
      <Span>
        <SvgIcon viewBox="0 0 24 18" style={{ width: 24, height: 18 }}>
          <rect width="24" height="2" />
          <rect y="8" width="24" height="2" />
          <rect y="16" width="24" height="2" />
        </SvgIcon>
      </Span>
    </Button>
  );
};
