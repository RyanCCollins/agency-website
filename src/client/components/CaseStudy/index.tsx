import * as React from 'react';
import { findDOMNode } from 'react-dom';
import { Box, Image, WithAnimation } from 'ui';
import Overlay from './overlay';
import Relative from './relative';

export interface State {
  isVisible: boolean;
}

export interface Props extends React.HTMLProps<typeof CaseStudy> {
  image: string;
  title: string;
}

export default class CaseStudy extends React.Component<Props, State> {
  constructor() {
    super();
    this.handleScroll = this.handleScroll.bind(this);
    this.state = {
      isVisible: false,
    };
  }
  public componentDidMount() {
    if (typeof window !== 'undefined') {
      this.handleScroll();
      window.addEventListener('scroll', this.handleScroll);
    }
  }
  public componentWillUnmount() {
    if (typeof window !== 'undefined') {
      window.removeEventListener('scroll', this.handleScroll);
    }
  }
  public divRef: Element;
  public handleScroll() {
    const windowHeight = window ? window.innerHeight : 1000;
    const node = findDOMNode(this.divRef);
    if (node) {
      const isVisible = node.getBoundingClientRect().top < windowHeight;
      this.setState({
        isVisible,
      });
    }
  }
  public render() {
    const { image, title } = this.props;
    return (
      <div ref={(ref) => { this.divRef = ref; }}>
        <WithAnimation type="fadeInUp" delay={100} {...this.state}>
          <Box pad="large">
            <Relative>
              <Overlay>
                <p>{title}</p>
              </Overlay>
              <Image imageSize="large" src={image} alt={title} />
            </Relative>
          </Box>
        </WithAnimation>
      </div>
    );
  }
}
