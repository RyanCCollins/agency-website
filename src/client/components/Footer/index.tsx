import * as React from 'react';
import { Footer as FooterComponent, Image, Box, Heading } from 'ui';

export default function Footer(): JSX.Element {
  return (
    <FooterComponent>
      <Box pad="medium" flexDirection="row" alignItems="center" justifyContent="space-between">
        <Image
          alt="The Agency"
          src="https://github.com/RyanCCollins/cdn/blob/master/misc/rocket.png?raw=true"
          imageSize="thumb"
        />
        <Box flexDirection="column">
          <Heading tag="h4">
            Â© 2017 The Agency
          </Heading>
        </Box>
      </Box>
    </FooterComponent>
  );
};
