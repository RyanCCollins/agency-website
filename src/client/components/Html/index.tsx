import * as React from 'react';
import serialize from 'serialize-javascript';

export interface Props {
  content: string;
  state: {};
  scriptHash: string;
  vendorHash: string;
  cssHash: string;
  styles: string;
}
export default function Html({
  content,
  state,
  scriptHash,
  vendorHash,
  cssHash,
  styles,
}: Props) {
  return (
    <html lang="en">
      <head>
        <meta charSet="UTF-8" />
        <meta httpEquiv="Content-Language" content="en" />
        <meta name="viewport" content="width=device-width,initial-scale=1" />
        <title>Scalable React TypeScript Boilerplate</title>
        <link href="https://fonts.googleapis.com/css?family=Hind|Montserrat|Roboto+Mono" rel="stylesheet" />
        <link href={`public${cssHash}`} rel="stylesheet" />
        <style dangerouslySetInnerHTML={{ __html: styles }} />
      </head>
      <body>
        <div id="app" dangerouslySetInnerHTML={{ __html: content }} />
        <script src={`public${scriptHash}`} charSet="UTF-8" />
        <script src={`public${vendorHash}`} type="text/javascript" />
        <script
          dangerouslySetInnerHTML={{
            __html: `window.__INITIAL_STATE__=${serialize(state, { isJSON: true })};`,
          }}
          charSet="UTF-8"
        />
      </body>
    </html>
  );
};
