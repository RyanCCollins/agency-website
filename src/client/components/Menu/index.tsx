import * as React from 'react';
import Component from './styles';

export interface Props extends React.HTMLProps<typeof Menu> {
  children: JSX.Element;
}
export default function Menu({
  children,
}: Props): JSX.Element {
  return (
    <Component>
      {children}
    </Component>
  );
};
