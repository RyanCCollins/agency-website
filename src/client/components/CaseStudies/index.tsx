import * as React from 'react';
import { Box } from 'ui';
import { CaseStudy } from 'components';
import { Project } from 'features/Portfolio/types';

export interface Props extends React.HTMLProps<typeof CaseStudies> {
  items: Project[];
}
export default function CaseStudies({
  items,
}: Props): JSX.Element {
  return (
    <Box flexDirection="column">
      {items.map((item, i) =>
        <CaseStudy key={i} {...item} />,
      )}
    </Box>
  );
};
