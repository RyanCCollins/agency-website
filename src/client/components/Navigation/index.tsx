import * as React from 'react';
import { Header, Box, Heading, Anchor } from 'ui';
import { Burger } from 'components';

export interface Props extends React.HTMLProps<typeof Navigation> {
  onToggleMenu: () => void;
  brandText: string;
}
export default function Navigation({
  brandText,
  onToggleMenu,
}: Props): JSX.Element {
  return (
    <Header>
      <Box
        style={{ width: '100%' }}
        flexDirection="row"
        alignItems="center"
        justifyContent="space-between"
      >
        <Anchor path="/" plain>
          <Heading margin="none" tag="h3" upcase>
            {brandText}
          </Heading>
        </Anchor>
        <Burger onClick={onToggleMenu} />
      </Box>
    </Header>
  );
};
