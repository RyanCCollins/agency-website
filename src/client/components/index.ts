/* GENERATOR-IMPORT */
import Html from './Html';

export {
  /* GENERATOR-EXPORT */
  Html,
};
