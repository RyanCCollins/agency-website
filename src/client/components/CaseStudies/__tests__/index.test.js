import { shallow, mount } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import React from 'react';
import { CaseStudy } from 'components';
import { Box } from 'ui';
import CaseStudies from '../index';
import props from './__mocks__/caseStudies.mock';

describe('<CaseStudies />', () => {
  describe('when shallow rendered', () => {
    const wrapper = shallow(
      <CaseStudies {...props} />,
    );
    it('should create a snapshot', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render a CaseStudy component for each item prop', () => {
      expect(wrapper.find(CaseStudy).length).toBe(props.items.length);
    });

    it('should render a wrapper Box component', () => {
      expect(wrapper.find(Box).length).toBe(1);
    });
  });

  describe('when mount rendered', () => {
    const wrapper = mount(
      <CaseStudies {...props} />,
    );
    it('should create a snapshot', () => {
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });
  });
});