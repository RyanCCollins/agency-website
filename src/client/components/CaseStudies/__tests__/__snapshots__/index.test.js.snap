// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseStudies /> when mount rendered should create a snapshot 1`] = `
<CaseStudies
  items={
    Array [
      Object {
        "__typename": "Project",
        "id": "58f43041f445ed0e26cd46e6",
        "image": "https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true",
        "title": "HPE Brand Central",
      },
      Object {
        "__typename": "Project",
        "id": "58f43041f445ed0e26cd46e7",
        "image": "https://github.com/RyanCCollins/cdn/blob/master/cms/bccmsmockup.jpg?raw=true",
        "title": "Grommet CMS",
      },
      Object {
        "__typename": "Project",
        "id": "58f43041f445ed0e26cd46e8",
        "image": "https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/datacentercaremockup.jpg?raw=true",
        "title": "HPE Datacenter Care",
      },
      Object {
        "__typename": "Project",
        "id": "58f43041f445ed0e26cd46ea",
        "image": "https://github.com/RyanCCollins/cdn/blob/master/portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true",
        "title": "Udacity Mentor Calendar",
      },
      Object {
        "__typename": "Project",
        "id": "58f43041f445ed0e26cd46e9",
        "image": "https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true",
        "title": "Astra Finance",
      },
    ]
  }
>
  <Box
    flexDirection="column"
    style={
      Object {
        "paddingBottom": 100,
      }
    }
  >
    <styled.div
      flexDirection="column"
      style={
        Object {
          "paddingBottom": 100,
        }
      }
    >
      <div
        className="lklsWn"
        style={
          Object {
            "paddingBottom": 100,
          }
        }
      >
        <CaseStudy
          __typename="Project"
          id="58f43041f445ed0e26cd46e6"
          image="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
          title="HPE Brand Central"
        >
          <withRouter(Anchor)
            path="/case-study/58f43041f445ed0e26cd46e6"
            plain={true}
          >
            <Anchor
              method="push"
              path="/case-study/58f43041f445ed0e26cd46e6"
              plain={true}
            >
              <styled.a
                onClick={[Function]}
                plain={true}
              >
                <a
                  className="ckTJUm"
                  onClick={[Function]}
                >
                  <div>
                    <WithAnimation
                      duration={1000}
                      isVisible={true}
                    >
                      <styled.div
                        delay={0}
                        duration={1000}
                        isVisible={true}
                        type="fadeIn"
                      >
                        <div
                          className="jOKFwR"
                          type="fadeIn"
                        >
                          <Box
                            pad="large"
                          >
                            <styled.div
                              pad="large"
                            >
                              <div
                                className="bWdHyO"
                              >
                                <styled.div>
                                  <div
                                    className="innYbI"
                                  >
                                    <styled.div>
                                      <div
                                        className="kPGFVC"
                                      >
                                        <p>
                                          HPE Brand Central
                                        </p>
                                      </div>
                                    </styled.div>
                                    <styled.img
                                      alt="HPE Brand Central"
                                      imageSize="large"
                                      src="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
                                      style={
                                        Object {
                                          "maxWidth": 700,
                                        }
                                      }
                                    >
                                      <img
                                        alt="HPE Brand Central"
                                        className="gcaPMe"
                                        src="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
                                        style={
                                          Object {
                                            "maxWidth": 700,
                                          }
                                        }
                                      />
                                    </styled.img>
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Box>
                        </div>
                      </styled.div>
                    </WithAnimation>
                  </div>
                </a>
              </styled.a>
            </Anchor>
          </withRouter(Anchor)>
        </CaseStudy>
        <CaseStudy
          __typename="Project"
          id="58f43041f445ed0e26cd46e7"
          image="https://github.com/RyanCCollins/cdn/blob/master/cms/bccmsmockup.jpg?raw=true"
          title="Grommet CMS"
        >
          <withRouter(Anchor)
            path="/case-study/58f43041f445ed0e26cd46e7"
            plain={true}
          >
            <Anchor
              method="push"
              path="/case-study/58f43041f445ed0e26cd46e7"
              plain={true}
            >
              <styled.a
                onClick={[Function]}
                plain={true}
              >
                <a
                  className="ckTJUm"
                  onClick={[Function]}
                >
                  <div>
                    <WithAnimation
                      duration={1000}
                      isVisible={true}
                    >
                      <styled.div
                        delay={0}
                        duration={1000}
                        isVisible={true}
                        type="fadeIn"
                      >
                        <div
                          className="jOKFwR"
                          type="fadeIn"
                        >
                          <Box
                            pad="large"
                          >
                            <styled.div
                              pad="large"
                            >
                              <div
                                className="bWdHyO"
                              >
                                <styled.div>
                                  <div
                                    className="innYbI"
                                  >
                                    <styled.div>
                                      <div
                                        className="kPGFVC"
                                      >
                                        <p>
                                          Grommet CMS
                                        </p>
                                      </div>
                                    </styled.div>
                                    <styled.img
                                      alt="Grommet CMS"
                                      imageSize="large"
                                      src="https://github.com/RyanCCollins/cdn/blob/master/cms/bccmsmockup.jpg?raw=true"
                                      style={
                                        Object {
                                          "maxWidth": 700,
                                        }
                                      }
                                    >
                                      <img
                                        alt="Grommet CMS"
                                        className="gcaPMe"
                                        src="https://github.com/RyanCCollins/cdn/blob/master/cms/bccmsmockup.jpg?raw=true"
                                        style={
                                          Object {
                                            "maxWidth": 700,
                                          }
                                        }
                                      />
                                    </styled.img>
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Box>
                        </div>
                      </styled.div>
                    </WithAnimation>
                  </div>
                </a>
              </styled.a>
            </Anchor>
          </withRouter(Anchor)>
        </CaseStudy>
        <CaseStudy
          __typename="Project"
          id="58f43041f445ed0e26cd46e8"
          image="https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/datacentercaremockup.jpg?raw=true"
          title="HPE Datacenter Care"
        >
          <withRouter(Anchor)
            path="/case-study/58f43041f445ed0e26cd46e8"
            plain={true}
          >
            <Anchor
              method="push"
              path="/case-study/58f43041f445ed0e26cd46e8"
              plain={true}
            >
              <styled.a
                onClick={[Function]}
                plain={true}
              >
                <a
                  className="ckTJUm"
                  onClick={[Function]}
                >
                  <div>
                    <WithAnimation
                      duration={1000}
                      isVisible={true}
                    >
                      <styled.div
                        delay={0}
                        duration={1000}
                        isVisible={true}
                        type="fadeIn"
                      >
                        <div
                          className="jOKFwR"
                          type="fadeIn"
                        >
                          <Box
                            pad="large"
                          >
                            <styled.div
                              pad="large"
                            >
                              <div
                                className="bWdHyO"
                              >
                                <styled.div>
                                  <div
                                    className="innYbI"
                                  >
                                    <styled.div>
                                      <div
                                        className="kPGFVC"
                                      >
                                        <p>
                                          HPE Datacenter Care
                                        </p>
                                      </div>
                                    </styled.div>
                                    <styled.img
                                      alt="HPE Datacenter Care"
                                      imageSize="large"
                                      src="https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/datacentercaremockup.jpg?raw=true"
                                      style={
                                        Object {
                                          "maxWidth": 700,
                                        }
                                      }
                                    >
                                      <img
                                        alt="HPE Datacenter Care"
                                        className="gcaPMe"
                                        src="https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/datacentercaremockup.jpg?raw=true"
                                        style={
                                          Object {
                                            "maxWidth": 700,
                                          }
                                        }
                                      />
                                    </styled.img>
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Box>
                        </div>
                      </styled.div>
                    </WithAnimation>
                  </div>
                </a>
              </styled.a>
            </Anchor>
          </withRouter(Anchor)>
        </CaseStudy>
        <CaseStudy
          __typename="Project"
          id="58f43041f445ed0e26cd46ea"
          image="https://github.com/RyanCCollins/cdn/blob/master/portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true"
          title="Udacity Mentor Calendar"
        >
          <withRouter(Anchor)
            path="/case-study/58f43041f445ed0e26cd46ea"
            plain={true}
          >
            <Anchor
              method="push"
              path="/case-study/58f43041f445ed0e26cd46ea"
              plain={true}
            >
              <styled.a
                onClick={[Function]}
                plain={true}
              >
                <a
                  className="ckTJUm"
                  onClick={[Function]}
                >
                  <div>
                    <WithAnimation
                      duration={1000}
                      isVisible={true}
                    >
                      <styled.div
                        delay={0}
                        duration={1000}
                        isVisible={true}
                        type="fadeIn"
                      >
                        <div
                          className="jOKFwR"
                          type="fadeIn"
                        >
                          <Box
                            pad="large"
                          >
                            <styled.div
                              pad="large"
                            >
                              <div
                                className="bWdHyO"
                              >
                                <styled.div>
                                  <div
                                    className="innYbI"
                                  >
                                    <styled.div>
                                      <div
                                        className="kPGFVC"
                                      >
                                        <p>
                                          Udacity Mentor Calendar
                                        </p>
                                      </div>
                                    </styled.div>
                                    <styled.img
                                      alt="Udacity Mentor Calendar"
                                      imageSize="large"
                                      src="https://github.com/RyanCCollins/cdn/blob/master/portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true"
                                      style={
                                        Object {
                                          "maxWidth": 700,
                                        }
                                      }
                                    >
                                      <img
                                        alt="Udacity Mentor Calendar"
                                        className="gcaPMe"
                                        src="https://github.com/RyanCCollins/cdn/blob/master/portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true"
                                        style={
                                          Object {
                                            "maxWidth": 700,
                                          }
                                        }
                                      />
                                    </styled.img>
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Box>
                        </div>
                      </styled.div>
                    </WithAnimation>
                  </div>
                </a>
              </styled.a>
            </Anchor>
          </withRouter(Anchor)>
        </CaseStudy>
        <CaseStudy
          __typename="Project"
          id="58f43041f445ed0e26cd46e9"
          image="https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true"
          title="Astra Finance"
        >
          <withRouter(Anchor)
            path="/case-study/58f43041f445ed0e26cd46e9"
            plain={true}
          >
            <Anchor
              method="push"
              path="/case-study/58f43041f445ed0e26cd46e9"
              plain={true}
            >
              <styled.a
                onClick={[Function]}
                plain={true}
              >
                <a
                  className="ckTJUm"
                  onClick={[Function]}
                >
                  <div>
                    <WithAnimation
                      duration={1000}
                      isVisible={true}
                    >
                      <styled.div
                        delay={0}
                        duration={1000}
                        isVisible={true}
                        type="fadeIn"
                      >
                        <div
                          className="jOKFwR"
                          type="fadeIn"
                        >
                          <Box
                            pad="large"
                          >
                            <styled.div
                              pad="large"
                            >
                              <div
                                className="bWdHyO"
                              >
                                <styled.div>
                                  <div
                                    className="innYbI"
                                  >
                                    <styled.div>
                                      <div
                                        className="kPGFVC"
                                      >
                                        <p>
                                          Astra Finance
                                        </p>
                                      </div>
                                    </styled.div>
                                    <styled.img
                                      alt="Astra Finance"
                                      imageSize="large"
                                      src="https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true"
                                      style={
                                        Object {
                                          "maxWidth": 700,
                                        }
                                      }
                                    >
                                      <img
                                        alt="Astra Finance"
                                        className="gcaPMe"
                                        src="https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true"
                                        style={
                                          Object {
                                            "maxWidth": 700,
                                          }
                                        }
                                      />
                                    </styled.img>
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Box>
                        </div>
                      </styled.div>
                    </WithAnimation>
                  </div>
                </a>
              </styled.a>
            </Anchor>
          </withRouter(Anchor)>
        </CaseStudy>
      </div>
    </styled.div>
  </Box>
</CaseStudies>
`;

exports[`<CaseStudies /> when shallow rendered should create a snapshot 1`] = `
<Box
  flexDirection="column"
  style={
    Object {
      "paddingBottom": 100,
    }
  }
>
  <CaseStudy
    __typename="Project"
    id="58f43041f445ed0e26cd46e6"
    image="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
    title="HPE Brand Central"
  />
  <CaseStudy
    __typename="Project"
    id="58f43041f445ed0e26cd46e7"
    image="https://github.com/RyanCCollins/cdn/blob/master/cms/bccmsmockup.jpg?raw=true"
    title="Grommet CMS"
  />
  <CaseStudy
    __typename="Project"
    id="58f43041f445ed0e26cd46e8"
    image="https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/datacentercaremockup.jpg?raw=true"
    title="HPE Datacenter Care"
  />
  <CaseStudy
    __typename="Project"
    id="58f43041f445ed0e26cd46ea"
    image="https://github.com/RyanCCollins/cdn/blob/master/portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true"
    title="Udacity Mentor Calendar"
  />
  <CaseStudy
    __typename="Project"
    id="58f43041f445ed0e26cd46e9"
    image="https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true"
    title="Astra Finance"
  />
</Box>
`;
